src/common/search/search-input.ts

```diff
@@ -7,33 +7,43 @@ import {
   property,
 } from "lit-element";
 import { fireEvent } from "../dom/fire_event";
-import "@polymer/iron-icon/iron-icon";
 import "@polymer/paper-input/paper-input";
 import "@polymer/paper-icon-button/paper-icon-button";
-import "@material/mwc-button";
+import "../../components/ha-icon";
+import { classMap } from "lit-html/directives/class-map";
 
 @customElement("search-input")
 class SearchInput extends LitElement {
   @property() public filter?: string;
+  @property({ type: Boolean, attribute: "no-label-float" })
+  public noLabelFloat? = false;
+  @property({ type: Boolean, attribute: "no-underline" })
+  public noUnderline = false;
 
   public focus() {
     this.shadowRoot!.querySelector("paper-input")!.focus();
   }
 
   protected render(): TemplateResult {
     return html`
+      <style>
```

Comment:
Why is this not in `styles` ?

Reply:
Because mixins don't work in `styles`

Reply:
Mixins shouldn't work at all 🤷‍♂ . The only reason they still work is because we have ShadyCSS on the page.

Anyway, we will have that as long as we have Polymer around, so no problem 👍